core/arch/arm/dts/stm32mp157c-ed1.dts

```diff
@@ -338,6 +338,7 @@
 };
 
 &rcc {
+	compatible = "st,stm32mp1-rcc-secure";
```

Comment:
That's weird, why not `compatible = "st,stm32mp1-rcc";` and `status = "disabled"; secure-status = "okay";`?

Reply:
Because playing with secure-status / status is not nice in the long term (e.g virtualized optee or other packages e.g. tf-m). Hence we prefer a specific compatible. This compatible has already been upstreamed to Linux kernel DT bindings for U-Boot/Linux to comply with:
https://github.com/torvalds/linux/blob/v6.6/Documentation/devicetree/bindings/clock/st%2Cstm32mp1-rcc.yaml#L59

Note: we have planned to propose a firewall resource framework in for device driver to possibly grant or query SoC firewall config. We've not proposed anything yet in OP-TEE but that on the todo list. There are discusions on this topic in the LKML, see latest [patch v6 series for STM32 firewall framework](https://lore.kernel.org/lkml/20231010125719.784627-1-gatien.chevallier@foss.st.com/) based on [Oleksii' patch for domain-controller](https://lore.kernel.org/lkml/cover.1668070216.git.oleksii_moisieiev@epam.com/).

Reply:
> Because playing with secure-status / status is not nice in the long term (e.g virtualized optee or other packages e.g. tf-m). Hence we prefer a specific compatible.

I do not really see which difference it makes, it is still the same hardware whether or not it is accessed from secure or non-secure world (which `status` and `secure-status` are for). But if the bindings have been accepted already that's fine. Thanks for context.